### DISCUSSION TITLE
Platform attribute vs. TestCase attribute

### DISCUSSION THREAD

**Discussion by tmx7777** [nunit/nunit#4289] [Q&A] [answered]
I recently found that the platform or target identifiers are different at [Platform] and [TestCase].

For example the syntax for [Platform] is`[Platform(Exclude = "Net-3.5")]`
and the syntax for [TestCase] is `[TestCase(ExcludePlatform = "net35")]`

The last one ("net35") is the official TFM (as I understand) and it does not work in the first case, and vice-versa.
I'm confused. Is this intentional? Did I understand something wrong here?

**Comment by manfred-brands** [comment, marked as answer]
Where did you find that? In the source code both attributes call into [PlatformHelper.IsPlatformSupported](https://github.com/nunit/nunit/blob/master/src/NUnitFramework/framework/Internal/PlatformHelper.cs#L76-L96)

Are you thinking of the .NET7 [SupportedOSPlatform](https://learn.microsoft.com/en-us/dotnet/api/system.runtime.versioning.supportedosplatformattribute?view=net-7.0)
Support for that was assed to the non-released NUnit4 only.

**Reply by tmx7777** [reply]
If I was unclear, I apologize. English is not my native language.  I did not find anything in the source code. I rather meant, I found out by trial and error using this feature. This is the behaviour on my system at least. I tried to use "Net-3.5" for [TestCase] and "net35" for [Platform], hoping one of them would work for both attributes. The same goes for "net48" and there is not even an identifier "Net-4.8" according to the documentation (https://docs.nunit.org/articles/nunit/writing-tests/attributes/platform.html). I'm working with NUnit3 and `nunit3-console.exe`.

In my projects I will have to deal with those legacy platforms for a while. This is managable however, tbh. I was just wondering.

Thanks for pointing out the call to IsPlatformSupported in the source. Looks pretty much straight forward. Yet, it seems to work differently on my system. I'll try to make a minimal reproducable example.

Thank you.

**Reply by tmx7777** [reply]
OK, turns out I wasn't careful enough and a little bit confused, too.

I guess the reason for my confusion was, that ...
- "Net-4.8" does not work, only "Net-4.5" works for .NET 4.8. 
- when using an invalid identifier (like "net35", "net48", or "xyz") the following happens:
  - when using it with [Platform], I get an error message "Invalid platform name",
  - when using it with [TestCase], **none** of the test cases are run and there is no error message and no entry in the internal trace either.
  
I'm sorry for the fuss and thank you for your help.
